dotCMS/src/main/java/com/dotcms/rest/api/v1/contenttype/ContentTypeResource.java

```diff
@@ -181,34 +188,39 @@ public Response updateType(@PathParam("id") final String id, final ContentTypeFo
 	@NoCache
 	@Produces({MediaType.APPLICATION_JSON, "application/javascript"})
 	public Response deleteType(@PathParam("id") final String id, @Context final HttpServletRequest req)
-			throws DotDataException, DotSecurityException, JSONException {
+			throws DotDataException, JSONException {
 
 		final InitDataObject initData = this.webResource.init(null, true, req, true, null);
 		final User user = initData.getUser();
 
 		ContentTypeAPI tapi = APILocator.getContentTypeAPI(user, true);
 
-
-		ContentType type = null;
 		try {
-			type = tapi.find(id);
-		} catch (NotFoundInDbException nfdb) {
+
+			ContentType type = null;
 			try {
 				type = tapi.find(id);
-			} catch (NotFoundInDbException nfdb2) {
-				return Response.status(404).build();
+			} catch (NotFoundInDbException nfdb) {
+				try {
+					type = tapi.find(id);
+				} catch (NotFoundInDbException nfdb2) {
+					return Response.status(404).build();
+				}
 			}
-		}
-
-		tapi.delete(type);
 
+			tapi.delete(type);
 
-		JSONObject joe = new JSONObject();
-		joe.put("deleted", type.id());
+			JSONObject joe = new JSONObject();
+			joe.put("deleted", type.id());
 
+			Response response = Response.ok(new ResponseEntityView(joe.toString())).build();
+			return response;
 
-		Response response = Response.ok(new ResponseEntityView(joe.toString())).build();
-		return response;
+		} catch (DotSecurityException e) {
+			throw new ForbiddenException(e);
+		} catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)